src/framework/components/model/component.js

```diff
@@ -539,6 +544,7 @@ Object.assign(pc, function () {
                             system.sphere = pc.createSphere(gd, {
                                 radius: 0.5
                             });
+                            system.sphere._refCount++;
```

Comment:
I'm not sure why this is needed. We shouldn't be accessing private properties in here.

Reply:
If you don't add 1 to the count here, the mesh will actually be released when the model destroy

Reply:
@daredevildave The only other way prevent a reference counted pc.Mesh from being blown away would be to create a dummy pc.MeshInstance and store that on the model component system. Alternatively, we could create a pc.Model for each primitive and then clone the pc.Model each instantiated primitive-based model component.

Reply:
> I'm not sure why this is needed. We shouldn't be accessing private properties in here.


_refCount is private, and the direct modification here is really not very good, if there is a better way to get the best

Reply:
> Alternatively, we could create a pc.Model for each primitive (on demand) and then clone the pc.Model for each instantiated primitive-based model component.

This sounds like a better approach to me.

Reply:
Yeah, I'd agree. It treats primitive-based models like asset-based models.

Reply:
> Yeah, I'd agree. It treats primitive-based models like asset-based models.
![image](https://user-images.githubusercontent.com/4427058/61920477-59e56980-af8c-11e9-9f94-31b5b1c733f9.png)
![image](https://user-images.githubusercontent.com/4427058/61920522-71bced80-af8c-11e9-9e7e-4c1405cc985b.png)
Is there a better way to create meshes in these two places? Is also the way to create a model and then pass clone?

Reply:
That probably should be how it's done, yes.